UltraStar Play/Assets/Scenes/Sing/SingSceneControl.cs

```diff
@@ -274,8 +295,230 @@ private void Start()
         {
             timeBarControl?.UpdateTimeValueLabel(songAudioPlayer.PositionInSongInMillis, songAudioPlayer.DurationOfSongInMillis);
         }));
+
+        // Modifiers apply callbacks
+        foreach (SingModifier modifier in sceneData.SingModifiers)
```

Comment:
- There is naturally quite a lot new logic in this class. I'd suggest to create a dedicated `PartyModeControl` for it. `SingSceneControl` can then delegate to this.